remote_mapper.go

```diff
@@ -77,12 +79,28 @@ func (m *RemoteMapper) Begin(c *influxql.Call, startingTime int64, chunkSize int
 		return err
 	}
 
-	// request to start streaming results
-	resp, err := http.Post(m.dataNodes[0].URL.String()+"/data/run_mapper", "application/json", bytes.NewReader(b))
-	if err != nil {
-		return err
+	var resp *http.Response
+	for {
+		node := m.dataNodes.Next()
+		if node == nil {
+			// no data nodes are available to service this query
+			return ErrShardNotFound
```

Comment:
This is a very important failure and deserves its own error code. Agreed? `ErrNoNodeAvailable` or something.